src/mbgl/renderer/painter.hpp

```diff
@@ -178,7 +177,7 @@ class Painter : private util::noncopyable {
     }();
 
 private:
-    TransformState state;
+    Transform& transform;
```

Comment:
I'm remembering why I did this now — this would require moving `transform.getLatLng()` down to the state as well, along with the spherical mercator math cache values. Seemed intrusive, but perhaps less so than the painter stuff. Trying this now.